logstash-core/src/main/java/org/logstash/common/clients/HttpClient.java

```diff
@@ -0,0 +1,656 @@
+
+                    httpClientBuilder.setSSLSocketFactory(sslSocketFactory);
+                } catch (Exception e) {
+                    throw new OptionsBuilderException("Unable to configure HTTP client with SSL/TLS", e);
+                }
+            }
+
+            if (usesBasicAuth()) {
+                String credentials = this.basicAuthUsername + ":" + this.basicAuthPassword;
+                String encodedCredentials = new String(Base64.getEncoder().encode(credentials.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
+
+                List<Header> headerList = Collections.singletonList(new BasicHeader("Authorization", "Basic " + encodedCredentials));
+                httpClientBuilder.setDefaultHeaders(headerList);
+            }
+
+            return new HttpClient(httpClientBuilder.build(), baseUrl);
+        }
+
+        private boolean usesBasicAuth() {
+            return (this.basicAuthUsername != null) && (this.basicAuthPassword != null);
+        }
+
+        private boolean usesSsl(URL baseUrl) {
+            return baseUrl.getProtocol().equals("https");
+        }
+
+        private static X509Certificate getCertificate(String certificateFilePath) throws OptionsBuilderException {
+            try (FileInputStream fis = new FileInputStream(certificateFilePath)) {
+                CertificateFactory cf = CertificateFactory.getInstance("X.509");
+                return (X509Certificate) cf.generateCertificate(fis);
+            } catch (FileNotFoundException e) {
+                throw new OptionsBuilderException("Could not read certificate at " + certificateFilePath, e);
+            } catch (IOException e) {
+                throw new OptionsBuilderException("Could not close certificate file at " + certificateFilePath + " after reading it", e);
+            } catch (Exception e) {
+                throw new OptionsBuilderException("Could not generate certificate from " + certificateFilePath, e);
+            }
+        }
+
+        private static RSAPrivateKey getPrivateKey(String privateKeyPath) throws OptionsBuilderException {
+            String privateKeyFileContents;
+            try {
+                privateKeyFileContents = new String(Files.readAllBytes(Paths.get(privateKeyPath)), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new OptionsBuilderException("Could not read private key file at " + privateKeyPath, e);
+            }
+
+            Matcher matcher = KEY_EXTRACTION_REGEXP.matcher(privateKeyFileContents);
+
+            final String obeMessage = "Could not parse private key file at " + privateKeyPath;
+
+            if (!matcher.matches()) {
+                throw new OptionsBuilderException(obeMessage);
+            }
+
+            String keyContentsBase64Encoded = matcher.group(2);
+            if (keyContentsBase64Encoded == null) {
+                throw new OptionsBuilderException(obeMessage);
+            }
+
+            byte[] keyContents;
```

Comment:
NIT: sorry for missing this earlier, but you can just move the declaration inline into the `try` here to clean things up a little.